homeassistant/components/zwave_js/api.py

```diff
@@ -278,3 +285,30 @@ async def remove_from_device_registry(
         return
 
     registry.async_remove_device(device.id)
+
+
+class DumpView(HomeAssistantView):
+    """View to dump the state of the Z-Wave JS server."""
+
+    url = "/api/zwave_js/dump"
+    name = "api:zwave_js:dump"
+
+    @RequestDataValidator({"config_entry_id": str})
+    async def post(self, request: web.Request, data: dict) -> web.Response:
+        """Dump the state of Z-Wave."""
+        hass = request.app["hass"]
+
+        if data["config_entry_id"] not in hass.data[DOMAIN]:
+            raise http_exceptions.HttpBadRequest
+
+        entry = hass.config_entries.async_get_entry(data["config_entry_id"])
```

Comment:
`async_get_entry` may return `None`. We may need to assert the entry for typing reasons.

Reply:
We already check it exists on the line above.

Reply:
Yes, I was thinking that mypy might complain about it since it can't follow that.